include/matx/transforms/fft.h

```diff
@@ -815,109 +911,209 @@ void fft(OutputTensor &o, const InputTensor &i,
 
   // Get parameters required by these tensors
-  auto params = detail::matxFFTPlan_t<OutputTensor, InputTensor>::GetFFTParams(o, i_new, 1);
+  auto params = detail::matxFFTPlan_t<decltype(out), decltype(in)>::GetFFTParams(out, in, 1);
   params.stream = stream;
 
   // Get cache or new FFT plan if it doesn't exist
   auto ret = detail::cache_1d.Lookup(params);
   if (ret == std::nullopt) {
-    auto tmp = new detail::matxFFTPlan1D_t<OutputTensor, InputTensor>{o, i_new};
+    auto tmp = new detail::matxFFTPlan1D_t<decltype(out), decltype(in)>{out, in};
     detail::cache_1d.Insert(params, static_cast<void *>(tmp));
-    tmp->Inverse(o, i_new, stream);
+    tmp->Inverse(out, in, stream);
   }
   else {
-    auto fft_type = static_cast<detail::matxFFTPlan1D_t<OutputTensor, InputTensor> *>(ret.value());
-    fft_type->Inverse(o, i_new, stream);
+    auto fft_type = static_cast<detail::matxFFTPlan1D_t<decltype(out), decltype(in)> *>(ret.value());
+    fft_type->Inverse(out, in, stream);
   }
+
+  if(!out.isSameView(o)) {
+    (o = out).run(stream);
+  }
+
+}
+
+/**
+ * Run a 1D IFFT with a cached plan
+ *
+ * Creates a new IFFT plan in the cache if none exists, and uses that to execute
+ * the 1D FFT. Note that FFTs and IFFTs share the same plans if all dimensions
+ * match
+ *
+ * @tparam OutputTensor
+ *   Output tensor or operator type
+ * @tparam InputTensor
+ *   Input tensor or operator type
+ * @tparam RANK
+ *   Rank of input and output tensors
+ * @param out
+ *   Output tensor or operator. The length of the fastest-changing dimension dictates the
+ * size of IFFT. If this size is longer than the length of the input tensor, the
+ * tensor will potentially be copied and zero-padded to a new block of memory.
+ * Future releases may remove this restriction to where there is no copy.
+ * @param in
+ *   input tensor or operator
+ * #param axis
+ *   axis to perform FFT along
+ * @param fft_size
+ *   Size of FFT. Setting to 0 uses the output size to figure out the FFT size.
+ * @param stream
+ *   CUDA stream
+ */
+template <typename OutputTensor, typename InputTensor>
+__MATX_INLINE__ void ifft(OutputTensor out, const InputTensor in, const int (&axis)[1], 
+         index_t fft_size = 0, cudaStream_t stream = 0)
+{
+  MATX_STATIC_ASSERT_STR(OutputTensor::Rank() == InputTensor::Rank(), matxInvalidDim,
+    "Input and output tensor ranks must match");  
+  
+  auto perm = detail::getPermuteDims<OutputTensor::Rank()>(axis);
+  ifft(permute(out, perm), permute(in, perm), fft_size, stream);
 }
 
+
 /**
  * Run a 2D FFT with a cached plan
  *
  * Creates a new FFT plan in the cache if none exists, and uses that to execute
  * the 2D FFT. Note that FFTs and IFFTs share the same plans if all dimensions
  * match
  *
- * @tparam T1
- *   Output view data type
- * @tparam T2
- *   Input view data type
+ * @tparam OutputTensor
+ *   Output operator or tensor
+ * @tparam InputTensor
+ *   Input operator or tensor
  * @tparam RANK
```

Comment:
RANK can be removed (I know this isn't yours)

Reply:
fixed